frontend/src/components/GamePhaseSpecificCenter.tsx

```diff
@@ -20,6 +21,17 @@ const GamePhaseSpecificCenter: React.FunctionComponent<Props> = ({
   dispatch,
   selectedUnit,
 }: Props) => {
+  useEffect(() => {
```

Comment:
This seems like a really weird place for this logic... Maybe just move this to a reducer or some custom middleware

Reply:
Good point, I though about the same. Will fix that

Reply:
Moved